db/migrate/20200520211047_adjust_chargeback_reports_features.rb

```diff
@@ -0,0 +1,52 @@
+class AdjustChargebackReportsFeatures < ActiveRecord::Migration[5.2]
+  class MiqProductFeature < ActiveRecord::Base; end
+  class MiqRolesFeature < ActiveRecord::Base; end
+
+  FEATURE_MAPPING = {
+    'chargeback_download_csv'  => 'chargeback_reports_download_csv',
+    'chargeback_download_pdf'  => 'chargeback_reports_download_pdf',
+    'chargeback_download_text' => 'chargeback_reports_download_text',
+    'chargeback_report_only'   => 'chargeback_reports_report_only'
+  }.freeze
+
+  def up
+    return if MiqProductFeature.none?
+
+    say_with_time 'Adjusting chargeback reports features for the non-explorer views' do
+      new_feature = MiqProductFeature.find_or_create_by!(:identifier => 'chargeback_reports_view')
+      MiqRolesFeature.where(:miq_product_feature_id => old_feature_ids).each do |feature|
+        MiqRolesFeature.create!(:miq_product_feature_id => new_feature.id, :miq_user_role_id => feature.miq_user_role_id)
+      end
```

Comment:
I don't think this logic is correct.  What I read here is that for _each_ of csv, pdf, text, and reports, create an extra roles-features link of chargeback_reports_view.  So, if the role has, say, all 4 features, then this will create 4 duplicate rows with the chargeback_reports_view in roles-features, which I think violates some uniqueness somewhere.

If you change your specs and attach more than one productfeature to a role you should see it fail.

I think the correct code would look something like:

user_role_ids_with_old_features = MiqRolesFeature.where(:miq_product_feature_id => old_feature_ids).pluck(:miq_user_role_id).uniq
user_role_ids_with_old_features.each do |user_roles_id|
  MiqRolesFeature.create!(:miq_product_feature_id => new_feature.id, :miq_user_role_id => user_role_id)
end
```

Reply:
@Fryguy you were right it did create multiple records for the new parent feature. Fixed migration and spec tests to verify changes, plus addressed https://github.com/ManageIQ/manageiq-schema/pull/487#discussion_r429449922